### DISCUSSION TITLE
Feature Request - Replied to snippet

### DISCUSSION THREAD

**Discussion by jlubeck** [flyerhq/flutter_chat_ui#97] [Ideas & Requests]
It is a normal use case to reply directly to a specific message, when this happens, it usually displays a snippet of the old message to give context.

Examples:
Telegram:
![IMG_6649](https://user-images.githubusercontent.com/3067603/118905553-eee7ee00-b8e9-11eb-9c57-6be4efd45986.jpg)

Whatsapp:
![IMG_6651](https://user-images.githubusercontent.com/3067603/118905547-ec859400-b8e9-11eb-9c2a-d436297bf70a.jpg)

**Comment by dariakhimych** [comment]
You're right @jlubeck, this is a nice feature to have in chats. And we already have it listed in our backlog 🤘

**Comment by jlubeck** [comment]
> You're right @jlubeck, this is a nice feature to have in chats. And we already have it listed in our backlog 🤘

Nice!

**Comment by jlubeck** [comment]
> You're right @jlubeck, this is a nice feature to have in chats. And we already have it listed in our backlog 🤘

I think I saw your public backlog somewhere but can't seem to find it... unless I'm thinking about another project. If you DO have it, would you mind adding it to the README.md ? Thanks!

**Comment by demchenkoalex** [comment]
Hey @jlubeck it is here https://click.flyer.chat/board but we are exploring use of GitHub instead of Notion ATM. We are not sure yet what we will pick. We will add this to the README when ready.

**Comment by jlubeck** [comment]
Thank you @demchenkoalex would definitely be awesome for the community to have everything in one place, but of course, it's your project 😄  Thank you for the link!

**Comment by jlubeck** [comment]
Hopefully this issue can be prioritized now that we left 1.1 behind? 😅 🙏 
I think it's the only missing link to have a somewhat full chat experience on a large group

**Comment by demchenkoalex** [comment]
We will see, right now React Native is a priority, it was left behind, so I will resolve small issues here until RN is done. Then we will continue with somewhat "big" features. But again PRs are welcome :) if needed, we can provide designs/UX for this feature here, so maybe somebody can take it.

**Comment by jlubeck** [comment]
> We will see, right now React Native is a priority, it was left behind, so I will resolve small issues here until RN is done. Then we will continue with somewhat "big" features. But again PRs are welcome :) if needed, we can provide designs/UX for this feature here, so maybe somebody can take it.

YES!! If you can provide design, I'm more than happy to try making a PR. The same goes for the other open issues :D

**Comment by demchenkoalex** [comment]
@dariakhimych your turn :)

**Comment by jlubeck** [comment]
@dariakhimych we are all counting on your blessings! (plus I can't code without designs :( )

**Comment by dariakhimych** [comment]
Hey @jlubeck 👋 Happy to share the brief where you will find some of Flyer design principles and guidelines as per how to better implement the Reply feature. In case of any questions do not hesitate to text me on Telegram. 

[Flyer Chat - Reply to a Message [Feature Brief].pdf](https://github.com/flyerhq/flutter_chat_ui/files/6726609/Flyer.Chat.-.Reply.to.a.Message.Feature.Brief.pdf)

**Comment by jlubeck** [comment]
Thank you for that @dariakhimych it's looking great! Only concern I have is if the X button on the reply is not too close to the send button of the input text?

**Comment by dariakhimych** [comment]
@jlubeck Please make sure you set a touch area for both at 40 points (required minimum) and they should be fine.

**Comment by dariakhimych** [comment]
Hey @jlubeck We're about to start implementing the Reply feature internally. Do you have any updates since we last spoke? Just assuring that we are not duplicating efforts. Any contributions you might have so far are welcomed. 🙏

**Reply by leeyisoft** [reply]
Has the message “Reply feature” been added and where can I see the code?

**Reply by demchenkoalex** [reply]
From the community - no. One of our devs is working on our implementation, it will be continued next year.

**Reply by Justus-M** [reply]
Hey @demchenkoalex  Glad to hear this is being implemented. Any ETA on this? Wondering whether I should wait or work on my own implementation

**Reply by demchenkoalex** [reply]
hey @Justus-M we almost had it and then war started (we are from Ukraine) so now I can't say for sure

**Reply by Justus-M** [reply]
@demchenkoalex Sorry to hear this! I hope everyone stays safe and healthy!

**Comment by Justus-M** [comment]
#213 I think scrolling to the original message when the reply is tapped would also be important for this. Ex. if the reply is to a message that is much further up in the chat, users can easily refer back to the original message (especially if it is truncated, or an image which can't properly be seen in the reply), as is the case in pretty much all popular chat apps.

Just mentioning this as I noticed it's not in the feature brief PDF shared above, and I think this is expected behavior for message replies in chat apps, so It will probably be highly requested.

**Reply by felixgabler** [reply]
Scrolling to a specific message (as long as it is loaded) will become possible with https://github.com/flyerhq/flutter_chat_ui/pull/209

**Comment by felixgabler** [comment]
Another functionality we need to think about is whether editing or deleting a message should be reflected in the replied to snippet. I believe users expect it to reflect the latest state of the message but that would be a bit more complicated with the data structures we have right now. To make this easier, we might want to save a reference to all replies inside the message. Then, when the message changes, users will have an easy way to update all replies. Of course this can already be implemented through metadata fields but perhaps it should be done as a fully supported concept. @demchenkoalex what do you think?

**Reply by demchenkoalex** [reply]
Content of the message you reply to should be updated when the original message changes, I think we covered it when developed an MVP, but I will double check. If original message is deleted, there will be no "replied" UI I guess.

**Reply by felixgabler** [reply]
What I meant was that the types don't have a field for this direction (i.e. store all replies to a message inside the message itself to change them when the message is updated/deleted). However, most databases should allow for querying the replies to a message to do exactly that without further fields in the message. So we should be fine.

**Reply by felixgabler** [reply]
As for what happens when the message is deleted, I think all replies should still show something. It could be confusing if it just removes the replied-to UI. This would be something for a designer to have a quick look. Perhaps just showing "Deleted" in the replied-to UI would be enough.

**Reply by Albert0sans** [reply]
Hello what is the sate of this?

**Reply by demchenkoalex** [reply]
Hello, I coded the whole package from scratch and right now closing on finishing all previously available features. Then I can code this.
